### PR TITLE
Added a new property called `LocaleClaimUri`  to identity.xml

### PR DESCRIPTION
**Pull request by ravindudsilva**
### Proposed changes in this pull request
This PR has added a new property called `LocaleClaimUri`  to identity.xml and updated the mapped value of `http://wso2.org/claims/locality`

Related issue - https://github.com/wso2/product-is/issues/12163

### CODE REVIEW

**Review by emswbandara — APPROVED** [review]

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 1]
```diff
@@ -1969,4 +1969,9 @@
         <ApplicationName>My Account</ApplicationName>
     </SystemApplications>
 
+    <!-- Claim to store localization code  -->
+    <!--    By setting true localization code will be stored inside the http://wso2.org/claims/locality-->
+    <!--    By setting false localization code will be stored inside the http://wso2.org/claims/local-->
+    <isLegacyLocalityClaimUri>true</isLegacyLocalityClaimUri>
```
Comment (emswbandara) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:1975]: shall we use to `<useLegacyLocalizationClaim>` ?

Comment (emswbandara) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:1975]: also the default value should be false.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -2762,4 +2762,10 @@
         	<Cookie name="{{cookie}}"/>
         	{% endfor %}
     </CookiesToInvalidate>
+
+    <!-- Claim to store localization code  -->
+    <!--    By setting true localization code will be stored inside the http://wso2.org/claims/locality-->
+    <!--    By setting false localization code will be stored inside the http://wso2.org/claims/local-->
+    <isLegacyLocalityClaimUri>true</isLegacyLocalityClaimUri>
```
Comment (emswbandara) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:2769]: let's make the value configurable from deployment.toml.
`<useLegacyLocalizationClaim>{{use_legacy_localization_claim}}</useLegacyLocalizationClaim>`

Comment (emswbandara) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:2769]: let's change this like follows.

```
{% if use_legacy_localization_claim is defined %}
        <useLegacyLocalizationClaim>{{use_legacy_localization_claim}}</useLegacyLocalizationClaim>
        {% else %}
        <useLegacyLocalizationClaim>false</useLegacyLocalizationClaim>
        {% endif %}
```

**Review by emswbandara — CHANGES_REQUESTED** [review]

**Review by IsuraD — APPROVED** [review]

**Comment by IsuraD** [comment]
We need to add this to the migration doc. Create a git issue and label migration.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 3]
```diff
@@ -1969,4 +1969,9 @@
         <ApplicationName>My Account</ApplicationName>
     </SystemApplications>
 
+    <!-- Claim to store localization code  -->
+    <!--    By setting true localization code will be stored inside the http://wso2.org/claims/locality-->
+    <!--    By setting false localization code will be stored inside the http://wso2.org/claims/local-->
+    <useLegacyLocalizationClaim>false</useLegacyLocalizationClaim>
```
[the remark this answers is not in the document]
Comment (ravindudsilva) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Fixed in commit 116c8e51105974b0f9fe5b87ec335f6176c37976

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 2 above)
Comment (ravindudsilva) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Fixed in commit 40f50964fb888adb7c5f78d42bd804eb164c13e8

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml (hunk 1 above)
Comment (ravindudsilva) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Fixed

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 4]
```diff
@@ -2762,4 +2762,14 @@
         	<Cookie name="{{cookie}}"/>
         	{% endfor %}
     </CookiesToInvalidate>
+
+    <!-- Claim to store localization code  -->
+    <!--    By setting true localization code will be stored inside the http://wso2.org/claims/locality-->
+    <!--    By setting false localization code will be stored inside the http://wso2.org/claims/local-->
+    {% if claims is defined && claims.use_legacy_localization_claim is defined %}
```
Comment (emswbandara) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:2769]: formatting issue.
lines 2770 to 2773 should be aligned with 2769

Comment (ravindudsilva) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:2770]: Fixed in commit 742fb3e45ddfa401035bfc2397f3aebf28b8cc20

**Review by IsuraD — APPROVED** [review]

**Review by emswbandara — APPROVED** [review]

**Pull request merged**